R/citsci_preprocess.R

```diff
@@ -56,12 +56,40 @@ make_tz_table <- function(){
     dplyr::mutate(timezone_lst = paste0("Etc/GMT+", (utc_offset_h * (-1))))
 }
 
+#' Get elevation based on lat/lon
+#'
+#' @return Elevation based on location
+#'
 get_elev <- function(lon_obs, lat_obs){
   locs = cbind(lon_obs, lat_obs)
   r = terra::rast("/vsicurl/https://prd-tnm.s3.amazonaws.com/StagedProducts/Elevation/13/TIFF/USGS_Seamless_DEM_13.vrt")
   terra::extract(r, locs) %>% as.numeric()
 }
 
+#' Geolocate location to assign state and ecoregion association
+#'
+#' @return State and ecoregion
+#'
+get_geoinfo <- function(lon_obs, lat_obs){
```

Comment:
@nayhur, let's split this into 3 functions:

- `get_eco_level3`
- `get_eco_level4`
- `get_state`

Or conversely, you could keep the generic `get_geoinfo` function and have the type of data (ecoregion level vs. state) be an argument to the function. We only want this function to return a single value or column of values at a time.